tests/test_browser.py

```diff
@@ -2331,7 +2331,7 @@ def test_runtime_misuse(self):
         xhr.open('GET', 'http://localhost:%s/report_result?' + HEAP32[Module.noted>>2]);
         xhr.send();
         setTimeout(function() { window.close() }, 1000);
-      }, 1000);
+      }, 2000);
```

Comment:
Can you make this a separate change? 

Makes me sad to see sleep values like this in test code.

Reply:
Yeah me too. I tried to fix it a different way, but couldn't get everything working, and didn't want to pollute this PR with unrelated stuff. I opened #8939, but I'm not really sure how to fix it.

I was hoping that increasing the time would make it pass, until a proper fix is implemented, but the tests still failed...

Reply:
Unless I am missing something, nothing else in the test depends on timing? So the value here could be 1, 1000, 2000, and the result should be the same. If so that makes me suspect it's failing for another reason.

Reply:
As far as I can see there's no guarantee that `myJSCallback()` will finish executing before this function is called. So the issue I was seeing was that `assert(Module.noted);` can be hit before it is set (or set to its final value). That's why it was was passing locally for me, but failing on the build system.
Also if `EXIT_RUNTIME=1` is specified, like it is here https://github.com/emscripten-core/emscripten/blob/30f701b18e0e377cca96c0fff36988e453b25486/tests/test_browser.py#L2353 then `myJSCallback()` could actually be called after the runtime has exited.

I think the best thing to do is fix #8939, merge the changes into this PR, then see if the tests still pass.

Reply:
I have a possible fix for that in #8964